include/aspect/particle/generator/uniform_radial.h

```diff
@@ -0,0 +1,112 @@
+/*
+ Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __aspect__particle_generator_uniform_radial_h
+#define __aspect__particle_generator_uniform_radial_h
+
+#include <aspect/particle/generator/interface.h>
+#include <aspect/simulator_access.h>
+
+#include <deal.II/base/std_cxx11/array.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      using namespace dealii;
+
+      /**
+       * Generate uniform radial distribution of particles over entire simulation domain.
+       */
+      template <int dim>
+      class UniformRadial : public Interface<dim>, public SimulatorAccess<dim>
```

Comment:
As for `UniformBox`, can you state what the difference between this class and `UniformRandom` is?

As a matter of fact, it is not clear what `uniform radial distribution` actually is. It suggests a probability density that only depends on radius. Is this correct? There is also the question whether you weigh the probability density with `r^d` or not, i.e., are you sampling uniformly in `r-theta` space (in 2d), or are you sampling uniformly in `x-y` space from a probability distribution that only depends on `r=sqrt(x^2+y^2)`?